pkg/proxy/config/api_test.go

```diff
@@ -203,26 +204,28 @@ func TestInitialSync(t *testing.T) {
 	defer close(stopCh)
 	sharedInformers.Start(stopCh)
 
-	err := wait.PollImmediate(time.Millisecond*10, wait.ForeverTestTimeout, func() (bool, error) {
-		svcHandler.lock.Lock()
-		defer svcHandler.lock.Unlock()
-		if reflect.DeepEqual(svcHandler.state, expectedSvcState) {
-			return true, nil
-		}
-		return false, nil
-	})
+	err := wait.PollUntilContextTimeout(context.Background(), time.Millisecond*10, wait.ForeverTestTimeout, true,
+		func(ctx context.Context) (bool, error) {
```

Comment:
This seems kind of pointless in the context of a unit test... was there any discussion about keeping simpler functions around for the tests? Or have we decided that we're just going to create unnecessary contexts everywhere?